src/model/include/iDynTree/ModelTestUtils.h

```diff
@@ -183,14 +200,25 @@ inline void getRandomJointPositions(VectorDynSize& vec, const Model& model)
             {
                 double max = jntPtr->getMaxPosLimit(i);
                 double min = jntPtr->getMinPosLimit(i);
-                vec(jntPtr->getDOFsOffset()+i) = getRandomDouble(min,max);
+                vec(jntPtr->getPosCoordsOffset()+i) = getRandomDouble(min,max);
             }
         }
         else
         {
-            for(int i=0; i < jntPtr->getNrOfPosCoords(); i++)
+            // Special handling for RevoluteSO2Joint (2 pos coords, 1 DOF)
+            if (jntPtr->getNrOfPosCoords() == 2 && jntPtr->getNrOfDOFs() == 1)
+            {
+                // Generate random angle and set complex representation
+                double angle = getRandomDouble(-M_PI, M_PI);
+                vec(jntPtr->getPosCoordsOffset()) = std::cos(angle);
+                vec(jntPtr->getPosCoordsOffset() + 1) = std::sin(angle);
+            }
+            else
             {
-                vec(jntPtr->getDOFsOffset()+i) = getRandomDouble();
+                for(int i=0; i < jntPtr->getNrOfPosCoords(); i++)
+                {
+                    vec(jntPtr->getPosCoordsOffset()+i) = getRandomDouble();
+                }
             }
         }
```

Comment:
I would just use the normalizeJointPos method here, to avoid have special case (also as it is possible that in the future we will have a 2 pos coords, 1 dof joint that is not a RevoluteSO2Joint  .